### DISCUSSION TITLE
How to attach to Console or Serial?

### DISCUSSION THREAD

**Discussion by onetr1ck** [cloud-hypervisor/cloud-hypervisor#3769] [Q&A]
Hi there,
iam currently running cloud hypervisor with --cmdline "console=hvc0" which uses tty by default. Since i want my vm to be started as a service, i need to be able to attach to the console at a later time. For this i tried "--console file=./myfile" and "socat ./myfile -". I can see the output here, but cannot interact. I couldn't find any example for attaching to console later from a different tty. I don't want to use screen.
Any thoughts on this?

**Comment by rbradford** [comment]
Using a PTY would be the best solution e.g. `--console pty`. You can find the path by using `ch-remote info`. You can then connect to that and interact as you wish.

**Comment by sboeuf** [comment]
The scenario you describe fits the PTY use case. Run you VM with `--serial off --console pty --api-socket /tmp/ch.sock`, once it runs, connect to the VM with `ch-remote` using the following command:
```
./ch-remote --api-socket=/tmp/ch.sock info
```

This will return the information about the PTY that has been created, which you can use to connect to it at a later time.

**Reply by maocili** [reply]
Can remotely controlled be supported in Windows?

**Reply by sboeuf** [reply]
If you're talking about RDP, yes you can reach a Windows guest with Remote Desktop Protocol given that you start the VM with a `--net`. Of course, this is assuming you prepared a Windows image with RDP and virtio-net enabled. You can refer to this documentation for more information: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/windows.md

**Reply by maocili** [reply]
I mean that remotely configure ip address with SAC. Also, I can start the VM with a -`-net tap=ich0 `. But I am fail to start the VM if add param `ip=`  .  My version is 22.0.

**Reply by sboeuf** [reply]
If you want access to SAC, run the VM with `--serial tty --console off`. The SAC will be provided through the serial.
